File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringBuilder sb = new StringBuilder();

        int n = Integer.parseInt(br.readLine());
        String arr[][] = new String[n][2];

        for (int i = 0; i < n; i++) {
            String input = br.readLine();
            arr[i][0] = input.split(" ")[0];
            arr[i][1] = input.split(" ")[1];
        }

        Arrays.sort(arr, ((o1, o2) -> {
            return Integer.parseInt(o1[0]) - Integer.parseInt(o2[0]);
        }));

        for (int i = 0; i < n; i++) {
            sb.append(arr[i][0] + " " +  arr[i][1]).append("\n");
        }

        bw.write(sb.toString());
        bw.flush();
        bw.close();
    }
}

File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val sb = StringBuilder()

    val n = br.readLine().toInt()
    val arr = ArrayList<Pair<Int, String>>()

    repeat(n) {
        val (age, name) = br.readLine().split(" ")
        arr.add(age.toInt() to name)
    }

    arr.sortWith(compareBy { it.first })

    for((name, age) in arr) {
        sb.append(name).append(" ").append(age).append("\n")
    }

    bw.write(sb.toString())
    bw.flush()
    bw.close()
}